draft-ietf-jsonpath-base.md

```diff
@@ -1229,6 +1230,34 @@ Using these operators to compare other types of values produces a "false" compar
 
 The semantics of regular expressions are as defined in {{-iregexp}}.
 
+The logical AND, OR, and NOT operators have the normal semantics of Boolean algebra and
+consequently obey these laws (where `P`, `Q`, and `R` are any expressions with syntax
+`logical-and-expr`, `T` is any true expression, such as `1 == 1`, and `F` is any false expression,
+such as `1 == 0`):
+
+| Law | Expression | Equivalent expression |
+|:--:|:--:|:--:|
+| Associativity of OR | `P || (Q || R)` | `(P || Q) || R`|
+| Associativity of AND | `P && (Q && R)` | `(P && Q) && R`|
+| Commutativity of OR | `P || Q` | `Q || R` |
+| Commutativity of AND | `P && Q` | `Q && R` |
+| Distributivity of OR over AND | `P || (Q && R)` | `(P || Q) && (P || R)`|
+| Distributivity of AND over OR | `P && (Q || R)` | `(P && Q) || (P && R)`|
+| Identity for OR | `P || F` | `P`|
+| Identity for AND | `P && T` | `P`|
+| Anihilator for OR | `P || T` | `T`|
+| Anihilator for AND | `P && F` | `F`|
+| Idempotency of OR | `P || P` | `P`|
+| Idempotency of AND | `P && P` | `P`|
+| Absorption 1 | `P && (P || Q)` | `P`|
+| Absorption 2 | `P || (P && Q)` | `P`|
+| Complementation 1 | `P && !(P)` | `F`|
+| Complementation 2 | `P || !(P)` | `T`|
+| Double negation | `!(!(P))` | `P`|
+| De Morgan 1 | `!(P) && !(Q)` | `!(P || Q)`|
+| De Morgan 2 | `!(P) || !(Q)` | `!(P && Q)`|
```

Comment:
Apparently, my review didn't go through and my comment was left unmade... 😒

Reply:
By "the above", do you mean the other laws above here in the table?

There may be a bit of redundancy. For example, I think (in the presence of the other laws) absorption 1 implies absorption 2 and vice versa, but it would be asymmetric to omit one of these on that basis.

It was tempting to omit the table and say that the operators obeyed the "usual" laws such as associativity, but that begged a pretty big question about what the "usual" laws are.

This is what happens when you have a closet mathematician as an editor...

Reply:
I'm not a _closet_ mathematician; it's my degree 😉.

Anyway, I was talking about _both_ De Morgan entries, but looking again, it appears they are defining distributivity of "not" `!`, which isn't really obvious from the other entries.

Really, a lot of these are derivable but not obvious. I guess they're with having so that readers don't have to do the work or make assumptions.

Meh, keep it in.  Ignore my comment.

Reply:
> I'm not a _closet_ mathematician; it's my degree wink.

Sounds like there are at least three of us with math(s) degrees.

> 
> Anyway, I was talking about _both_ De Morgan entries, but looking again, it appears they are defining distributivity of "not" `!`, which isn't really obvious from the other entries.

Distributivity of NOT would make, for example, `!(P) && !(Q)` equivalent to `!(P && Q)`, which is incorrect.

> 
> Really, a lot of these are derivable but not obvious. I guess they're [worth] having so that readers don't have to do the work or make assumptions.
> 
> Meh, keep it in. Ignore my comment.

Thanks.